lib/onelogin/api/cursor.rb

```diff
@@ -55,16 +60,21 @@ def fetch_next_page
       query: @params
     )
 
-    json = response.parsed_response
-    results = json['data'].flatten
+    if response.code == 200
+      json = response.parsed_response
+      results = json['data'].flatten
+
+      @collection += if results_remaining < results.size
+        results.slice(0, results_remaining)
+      else
+        results
+      end
 
-    @collection += if results_remaining < results.size
-      results.slice(0, results_remaining)
+      @after_cursor = after_cursor(json)
     else
-      results
+      @error = response.code.to_s
```

Comment:
This doesnt actually work. If you run this branch in the console using some Auth Only credentials you would expect it to produce a 401 error but in reality it just returns nil. 

e.g. Run the sdk console
```sh
> bin/console
```

```ruby
client = OneLogin::Api::Client.new(
  client_id: 'xxx',
  client_secret:'xxx',
  region: 'us'
)

users = client.get_users
# returns the cursor but no error yet as the users not fetched until each is called

users = client.get_users.each{|u|u.id}
# returns nil and since the error is applied to the cursor and not the client it is not visible 
```